docs: Add RapidAPI guide (#2015)
apify/apify-docs: sources/academy/tutorials/apify_actors/adding_rapidapi_project.mdx

---
title: Adding your RapidAPI project to Apify
description: If you've already published an API project on RapidAPI and want to reach a broader audience, Apify Store and its active user community can help you expand your reach. You can maintain your RapidAPI presence while tapping into the Apify ecosystem.
sidebar_position: 1
category: apify platform
slug: /apify-actors/adding-rapidapi-project
---

If you've already published an API project on RapidAPI and want to reach a broader audience, Apify Store and its active user community can help you expand your reach. You can maintain your RapidAPI presence while tapping into the Apify ecosystem.

---

## Why add your project to Apify?

By publishing your API on Apify, you'll reach thousands of active users in Apify Store. You'll also get access to the Apify platform's infrastructure: managed hosting, data storage, scheduling, advanced web scraping and crawling capabilities, and integrated proxy management. These tools help you reach more users and enhance your API's functionality.

## Step-by-step guide

> This example uses Express.js, but these steps work for any Node.js or Python API framework - just adapt the code to your stack.

Reviewer: I'd phrase it something like this:
```suggestion
We'll demonstrate the approach on an app built on top of [Express.js](https://expressjs.com/), but with a few adaptations to the code, any API framework will work.
```

You'll deploy your API as an [Apify Actor](https://apify.com/actors) - a serverless cloud program that runs on the Apify platform. Actors can handle everything from simple automation to running web servers.

### Prerequisites

You’ll need an [Apify account](https://console.apify.com/sign-in) - **it’s free and no credit card is required**. For simple migration and deployment, we recommend installing Apify CLI:

```bash
curl -fsSL https://apify.com/install-cli.sh | bash
```

:::info CLI installation

For more information, check the [CLI installation page](https://docs.apify.com/cli/docs/installation).

:::

### Step 1: Initialize the Actor structure

> Once you have Apify CLI, you can run the following command:

Reviewer: ```suggestion
Once you have the Apify CLI, run the following command:
```

### Step 2: Add Actor logic

The first important thing is the initialization of the Actor. The second is the correct mapping of the PORT. Check the following example for inspiration:

```jsx
await Actor.init(); // Initializes the Actor

const app = express();
const PORT = Actor.config.get('containerPort'); // Specifies the PORT
const DATA_FILE = path.join(__dirname, 'data', 'items.json');

app.use(express.json());

// Rest of the logic
```

**Tip:** The Apify platform performs readiness checks by sending GET requests to `/` with the `x-apify-container-server-readiness-probe` header. For better resource efficiency, consider checking for this header and returning a simple response early, rather than processing it as a full request. This optimization is particularly useful for resource-intensive Actors.

```jsx
app.get('*', (req, res) => {
    if (req.headers['x-apify-container-server-readiness-probe']) {
        console.log('Readiness probe');
        res.send('Hello, readiness probe!\n');
    }
});
```

### Step 3: Test your Actor locally

> Once you’ve added the Actor logic, you can test your Actor locally with the following command:

Reviewer: ```suggestion
Once you’ve added the Actor logic, test your Actor locally with the following command:
```

> ```jsx

Reviewer: ```suggestion
```bash
```

sidebar_position: 1
category: apify platform
slug: /apify-actors/adding-rapidapi-project
---

If you've already published an API project on [RapidAPI](https://rapidapi.com/) and want to reach a broader audience, Apify Store and its active user community can help you expand your reach. You can maintain your RapidAPI presence while tapping into the Apify ecosystem.

---

## Why add your project to Apify?

By publishing your API on Apify, you'll reach thousands of active users in Apify Store. You'll also get access to the Apify platform's infrastructure: managed hosting, data storage, scheduling, advanced web scraping and crawling capabilities, and integrated proxy management. These tools help you reach more users and enhance your API's functionality.

## Step-by-step guide

We'll demonstrate the approach on an app built on top of [Express.js](https://expressjs.com/), but with a few adaptations to the code, any API framework will work.

You'll deploy your API as an [Apify Actor](https://apify.com/actors) - a serverless cloud program that runs on the Apify platform. Actors can handle everything from simple automation to running web servers.

### Prerequisites

> You’ll need an [Apify account](https://console.apify.com/sign-in) - **it’s free and no credit card is required**. For simple migration and deployment, we recommend installing the Apify CLI:

Reviewer 2: If we really need to emphasize we use italics for this. Bold is reserved for UI elements

```suggestion
You’ll need an [Apify account](https://console.apify.com/sign-in) - _it’s free and no credit card is required_. For simple migration and deployment, we recommend installing the Apify CLI:
```
